application/shared-kernel/ApiCore/Endpoints/TrackEndpoints.cs

```diff
@@ -0,0 +1,195 @@
+                        trackRequestDto.Data.BaseData.Properties, new Dictionary<string, double>())
+                    {
+                        SeverityLevel = trackRequestDto.Data.BaseData.SeverityLevel,
+                        Timestamp = DateTimeOffset.Parse(trackRequestDto.Time)
+                    };
+
+                    CopyDictionary(trackRequestDto.Data.BaseData.Properties, telemetry.Properties);
+                    CopyDictionary(trackRequestDto.Data.BaseData.Measurements, telemetry.Metrics);
+
+                    telemetryClient.TrackException(telemetry);
+                    break;
+                }
+                case "MetricData":
+                {
+                    var metric = trackRequestDto.Data.BaseData.Metrics.First();
+                    var telemetry = new MetricTelemetry
+                    {
+                        Name = metric.Name,
+                        Sum = metric.Value,
+                        Count = metric.Count,
+                        Timestamp = DateTimeOffset.Parse(trackRequestDto.Time)
+                    };
+
+                    CopyDictionary(trackRequestDto.Data.BaseData.Properties, telemetry.Properties);
+
+                    telemetryClient.TrackMetric(telemetry);
+                    break;
+                }
+                case "RemoteDependencyData":
+                    // Ignore remote dependency data
+                    break;
+                default:
+                    logger.LogWarning($"Unsupported telemetry type: {trackRequestDto.Data.BaseType}");
+                    break;
+            }
+        }
+
+        return new TrackResponseSuccessDto(true, "Telemetry sent.");
+    }
+
+    private static IEnumerable<ExceptionDetailsInfo> GetExceptionDetailsInfos(TrackRequestDto trackRequestDto)
+    {
+        var exceptionDetailsInfos = trackRequestDto.Data.BaseData.Exceptions
+            .Select(exception => new ExceptionDetailsInfo(
+                0,
+                0,
+                exception.TypeName,
+                exception.Message,
+                exception.HasFullStack,
+                exception.Stack,
+                exception.ParsedStack.Select(parsedStack => new StackFrame(
+                    parsedStack.Assembly,
+                    parsedStack.FileName,
+                    parsedStack.Level,
+                    parsedStack.Line,
+                    parsedStack.Method
+                ))
+            ));
+        return exceptionDetailsInfos;
+    }
+
+    private static void CopyDictionary<TValue>(IDictionary<string, TValue>? source, IDictionary<string, TValue> target)
+    {
+        if (source == null) return;
+
+        foreach (var pair in source)
+        {
+            if (string.IsNullOrEmpty(pair.Key))
+            {
+                continue;
+            }
+
+            if (!target.ContainsKey(pair.Key))
+            {
+                target[pair.Key] = pair.Value;
+            }
+        }
```

Comment:
Use `is null` over `== null` and flatten structure:

```suggestion
        if (source is null) return;

        foreach (var pair in source)
        {
            if (string.IsNullOrEmpty(pair.Key)) continue;
            if (target.ContainsKey(pair.Key)) continue;
            target[pair.Key] = pair.Value;
        }
```